src/routes/query/search.ts

```diff
@@ -7,9 +8,38 @@ import {
 
 export default defineEventHandler(async (event) => {
   // Parse the request
-  const { userId, query, limit, excludeNodeTypes } =
+  const { userId, query, limit, excludeNodeTypes, conversationId } =
     querySearchRequestSchema.parse(await readBody(event));
-  return querySearchResponseSchema.parse(
-    await searchMemory({ userId, query, limit, excludeNodeTypes }),
-  );
+
+  // Get the standard search results
+  const standardResults = await searchMemory({ 
+    userId, 
+    query, 
+    limit, 
+    excludeNodeTypes 
+  });
+
+  // If no conversationId is provided, just return standard results
+  if (!conversationId) {
+    return querySearchResponseSchema.parse(standardResults);
+  }
+
+  // Try to get deep research results from cache
+  const deepResults = await getDeepResearchResult(userId, conversationId);
+
+  // If no deep research results, return standard results
+  if (!deepResults) {
+    return querySearchResponseSchema.parse(standardResults);
+  }
+
+  // Merge standard and deep research results
+  // If both have content, combine them with standard results first
+  const combinedResults: QuerySearchResponse = {
+    query,
+    formattedResult: standardResults.formattedResult
```

Comment:
As mentioned above, I think it's better if we can run a single formatting of all the results combined. Probably we need to refactor searchMemory a bit too so the formatting is separate from the searching.